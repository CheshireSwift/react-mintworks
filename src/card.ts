export enum CardType {
  Culture,
  Production,
  Utility,
  Deed
}

export type CardConfig = {
  name: string;
  cost: number;
  picture: string;
  effect: React.ReactNode;
  score: number;
  type: CardType;
};

export const typeConfig = {
  [CardType.Culture]: { icon: 'herb', color: 'yellowgreen' },
  [CardType.Production]: { icon: 'gear', color: 'indianred' },
  [CardType.Utility]: { icon: 'wrench', color: 'gold' },
  [CardType.Deed]: { icon: 'scroll', color: 'paleturquoise' }
};
