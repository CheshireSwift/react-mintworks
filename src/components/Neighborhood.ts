import * as R from 'ramda';
import { div, h } from 'react-hyperscript-helpers';
import emoji from 'node-emoji';

import { CardConfig, typeConfig } from '../card';

type NeighborhoodProps = {
  cards: CardConfig[];
};

const BuildingIcon = (props: CardConfig) =>
  div(
    {
      key: props.name,
      style: {
        textAlign: 'center',
        backgroundColor: typeConfig[props.type].color,
        width: '4rem',
        height: '4rem',
        padding: '0.3rem',
        display: 'flex',
        flexDirection: 'column',
        justifyContent: 'space-around',
        margin: '0.25rem'
      }
    },
    [
      div({ style: { fontSize: 24 } }, props.picture),
      div(
        { style: { fontSize: 10, wordBreak: 'break-all' } },
        R.repeat(emoji.get('star'), props.score)
      )
    ]
  );

export const Neighborhood = ({ cards }: NeighborhoodProps) =>
  div(
    {
      style: {
        width: '10rem',
        backgroundColor: 'grey',
        padding: '0.3rem',
        display: 'flex',
        flexWrap: 'wrap',
        border: '0.2rem dashed white'
      }
    },
    cards.map(card => h(BuildingIcon, card))
  );

export default Neighborhood;
