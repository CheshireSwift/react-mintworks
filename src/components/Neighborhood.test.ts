import { render } from 'enzyme';
import { h } from 'react-hyperscript-helpers';
import emoji from 'node-emoji';
import * as R from 'ramda';

import { CardType } from '../card';

import Neighborhood from './Neighborhood';

describe('the neighborhood', () => {
  const stars = (n: number) => R.repeat(emoji.get('star'), n).join('');

  const card = {
    name: 'Mine',
    cost: 2,
    picture: emoji.get('pick'),
    effect: '*Upkeep:* Gain ()',
    score: 2,
    type: CardType.Production
  };

  const cards = [card];

  const neighborhoodText = render(h(Neighborhood, { cards })).text();

  it('matches the snapshot', () => {
    expect(render(h(Neighborhood, { cards }))).toMatchSnapshot();
  });

  it('displays the buildings', () => {
    expect(neighborhoodText).toContain(card.picture);
  });

  it('displays the score for each building', () => {
    expect(neighborhoodText).toContain(stars(2));
    expect(neighborhoodText).not.toContain(stars(3));
  });
});
