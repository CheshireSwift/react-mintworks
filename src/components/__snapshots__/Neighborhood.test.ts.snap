// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the neighborhood matches the snapshot 1`] = `
<div
  style="width:10rem;background-color:grey;padding:0.3rem;display:flex;flex-wrap:wrap;border:0.2rem dashed white"
>
  <div
    style="text-align:center;background-color:indianred;width:4rem;height:4rem;padding:0.3rem;display:flex;flex-direction:column;justify-content:space-around;margin:0.25rem"
  >
    <div
      style="font-size:24px"
    >
      ⛏️
    </div>
    <div
      style="font-size:10px;word-break:break-all"
    >
      ⭐⭐
    </div>
  </div>
</div>
`;
